packages/cli/src/commands/rewards/show.ts

```diff
@@ -0,0 +1,261 @@
+
+  async run() {
+    const res = this.parse(Show)
+    const election = await this.kit.contracts.getElection()
+    const validators = await this.kit.contracts.getValidators()
+    const epochSize = await validators.getEpochSize()
+    const currentBlock = await this.web3.eth.getBlockNumber()
+    const lastEpochBlock = Math.floor(currentBlock / epochSize) * epochSize
+    const fromBlock: number = lastEpochBlock - epochSize * ((res.flags.epochs || 1) - 1)
+    const address = res.flags.address || ''
+    let voterRewards: VoterReward[] = []
+    let validatorRewards: ValidatorReward[] = []
+    let validatorGroupRewards: ValidatorGroupReward[] = []
+
+    if (address) {
+      await newCheckBuilder(this)
+        .isAccount(address)
+        .runChecks()
+    }
+
+    // For each epoch...
+    for (let blockNumber = fromBlock; blockNumber <= lastEpochBlock; blockNumber += epochSize) {
+      // Get the groups that address voted for at this epoch, and those groups' total votes.
+      const addressVotes: { [key: string]: BigNumber } = {}
+      const totalGroupVotes: { [key: string]: BigNumber } = {}
+      if (address) {
+        const voter = await election.getVoter(address, blockNumber)
+        for (const vote of voter.votes) {
+          const group: string = vote.group.toLowerCase()
+          addressVotes[group] = (addressVotes[group] || new BigNumber(0)).plus(vote.active)
+          totalGroupVotes[group] = (totalGroupVotes[group] || new BigNumber(0)).plus(
+            await election.getTotalVotesForGroup(group, blockNumber)
+          )
+        }
+      }
+
+      // voterRewardsEvent applies to address when voterRewardsEvent.group in addressVotes.
+      const epochVoterRewardsEvents = await election.getPastEvents(
+        'EpochRewardsDistributedToVoters',
+        {
+          fromBlock: blockNumber,
+          toBlock: blockNumber,
+        }
+      )
+
+      const voterRewardsValidatorGroupDetails: ValidatorGroup[] = await Promise.all(
+        epochVoterRewardsEvents.map((e: EventLog) =>
+          validators.getValidatorGroup(e.returnValues.group)
+        )
+      )
+
+      const epochVoterRewards = epochVoterRewardsEvents.map(
+        (e: EventLog, index: number): VoterReward => ({
+          name: voterRewardsValidatorGroupDetails[index].name,
+          group: e.returnValues.group,
+          value: e.returnValues.value,
+          blockNumber: e.blockNumber,
+          addressVotes: addressVotes[e.returnValues.group],
+          totalGroupVotes: totalGroupVotes[e.returnValues.group],
+        })
+      )
+
+      voterRewards = voterRewards.concat(
+        address
+          ? epochVoterRewards.filter((e: VoterReward) => e.group.toLowerCase() in addressVotes)
+          : epochVoterRewards
+      )
+
+      // validatorRewardEvent applies to address when validatorRewardEvent.validator is address.
+      const epochValidatorRewardsEvents = await validators.getPastEvents(
```

Comment:
same

Reply:
The reason is that the end user doesn't really know a lot fo types of events, and event parameters. We need to digest that for the end user